test/module/irohad/ametsuchi/postgres_block_storage_test.cpp

```diff
@@ -0,0 +1,161 @@
+  shared_model::crypto::Blob blob_ = shared_model::crypto::Blob(
+      shared_model::crypto::Blob::Bytes{0, 1, 5, 17, 66, 255});
+  std::string creator_ = "user1@test";
+};
+
+/**
+ * @given initialized block storage, single block with height_ inserted
+ * @when another block with height_ is inserted
+ * @then second insertion fails
+ */
+TEST_F(PostgresBlockStorageTest, InsertTest) {
+  ASSERT_TRUE(block_storage_->insert(mock_block_));
+  ASSERT_FALSE(block_storage_->insert(mock_block_));
+}
+
+/**
+ * @given initialized block storage, single block with height_ inserted
+ * @when block with height_ is fetched
+ * @then it is returned
+ */
+TEST_F(PostgresBlockStorageTest, FetchExisting) {
+  auto tx = TestTransactionBuilder().creatorAccountId(creator_).build();
+  std::vector<shared_model::proto::Transaction> txs;
+  txs.push_back(std::move(tx));
+  auto block = TestBlockBuilder().height(height_).transactions(txs).build();
+
+  ASSERT_TRUE(block_storage_->insert(clone(block)));
+
+  auto block_var = *(block_storage_->fetch(block.height()));
+  ASSERT_EQ(block.blob(), block_var->blob());
+}
+
+/**
+ * @given initialized block storage without blocks
+ * @when block with height_ is fetched
+ * @then nothing is returned
+ */
+TEST_F(PostgresBlockStorageTest, FetchNonexistent) {
+  ASSERT_FALSE(block_storage_->fetch(height_));
+}
+
+/**
+ * @given initialized block storage, single block with height_ inserted
+ * @when size is fetched
+ * @then 1 is returned
+ */
+TEST_F(PostgresBlockStorageTest, Size) {
+  ASSERT_TRUE(block_storage_->insert(mock_block_));
+  ASSERT_EQ(1, block_storage_->size());
+}
+
+/**
+ * @given initialized block storage, single block with height_ inserted
+ * @when storage is cleared with clear
+ * @then no blocks are left in storage
+ */
+TEST_F(PostgresBlockStorageTest, Clear) {
+  ASSERT_TRUE(block_storage_->insert(mock_block_));
+  block_storage_->clear();
+  ASSERT_FALSE(block_storage_->fetch(height_));
+  ASSERT_EQ(0, block_storage_->size());
+}
+
+/**
+ * @given initialized block storage, two blocks with height_ and height_+2 are
+ * inserted
+ * @when forEach is called
+ * @then both blocks are visited, lambda is invoked twice
+ */
+TEST_F(PostgresBlockStorageTest, ForEach) {
+  auto tx = TestTransactionBuilder().creatorAccountId(creator_).build();
+  std::vector<shared_model::proto::Transaction> txs;
+  txs.push_back(std::move(tx));
+  auto block = TestBlockBuilder().height(height_).transactions(txs).build();
+  auto another_block =
+      TestBlockBuilder().height(height_ + 2).transactions(txs).build();
+
+  ASSERT_TRUE(block_storage_->insert(clone(block)));
+  ASSERT_TRUE(block_storage_->insert(clone(another_block)));
+
+  size_t count = 0;
+
+  block_storage_->forEach([&count, &block, &another_block](const auto &b) {
+    ++count;
+    if (b->height() == block.height()) {
+      ASSERT_EQ(b->blob(), block.blob());
+    } else if (b->height() == another_block.height()) {
```

Comment:
does the order matter here?
also, you could benifit from using a vector of blocks and index addressing into it, but that's not important since you've already done another way.

Reply:
I think the order is not important here. It will be probably saved but we just don't care.
I agree about vector but there are just 2 blocks and is looks reasonable not to pack them inside a collection.